src/main.rs

```diff
@@ -6,89 +6,47 @@
 // copied, modified, or distributed except according to those terms.
 
 extern crate cargo_metadata;
-extern crate docopt;
+extern crate clap;
 extern crate rustc_serialize;
 extern crate term;
 
 use cargo_metadata::{metadata, Package};
-use docopt::Docopt;
+use clap::{App, Arg, SubCommand, ArgMatches, AppSettings};
 use std::{env, error, fs, io, path, process};
 use std::io::Write;
 
 mod utils;
 
-const USAGE: &'static str = "
-Cargo Fuzz
-
-Usage:
-  cargo fuzz --init
-  cargo fuzz --fuzz-target TARGET
-  cargo fuzz --add TARGET
-  cargo fuzz --list
-  cargo fuzz (-h | --help)
-
-Options:
-  -h --help              Show this screen.
-  --init                 Initialize fuzz folder
-  --fuzz-target TARGET   Run with given fuzz target in fuzz/fuzzers
-  --add TARGET           Add a new fuzz target
-  --list                 List the available fuzz targets
-";
-
-#[derive(Debug, RustcDecodable)]
-struct Args {
-    flag_init: bool,
-    flag_add: Option<String>,
-    flag_fuzz_target: Option<String>,
-    flag_list: bool,
-}
-
 fn main() {
-    let args: Args = Docopt::new(USAGE)
-                            .and_then(|d| d.decode())
-                            .unwrap_or_else(|e| e.exit());
-
-    let result = if args.flag_init {
-        init_fuzz()
-    } else if let Some(target) = args.flag_add {
-        add_target(target)
-    } else if let Some(target) = args.flag_fuzz_target {
-        let result = run_target(target);
-        if let Ok(success) = result {
-            if success {
-                // Can this ever happen?
-                Ok(())
-            } else {
-                utils::print_message("Fuzzing found errors!", term::color::YELLOW);
```

Comment:
The fuzzer binary is now `exec`d. Moreover I’m not sure about proclaiming "fuzzing" found errors based purely on the fact that binary exit with non-0 code too. We’ve observed at least multiple cases where this could be true in practice:

* OOM (not necessarily an error uncovered by fuzzing);
* Bugs in the fuzzer itself.

So instead of claiming something, I just made it claim nothing, also getting rid of some small over-head in process.